chore: address compiler warnings

github-roushan opened the pull request:
## Describe Your Changes
- This pull request addresses a range of compiler warnings across multiple files, improving code quality and maintainability.
- It implements handling for `[[nodiscard]]` attributes where return values were previously being ignored, ensuring proper error checking and resource management.
- Added comprehensive error checking and logging to provide better diagnostics and identify potential issues during runtime.

## Fixes Issues

- Closes #
- Closes #

## Self Checklist

- [x] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

github-roushan committed:
chore:  resolve warnings, handle nodiscard, and add error logging

github-roushan commented:
@vansangpfiev

vansangpfiev commented on engine/utils/command_executor.h:
```diff
@@ -45,5 +45,5 @@ class CommandExecutor {
 
  private:
-  std::unique_ptr<FILE, decltype(&PCLOSE)> m_pipe{nullptr, PCLOSE};
+ std::unique_ptr<FILE, void (*)(FILE*)> m_pipe{nullptr, [](FILE* file) { if (file) { pclose(file); } }};
```
Please check the compilation issue on Windows
```
C:\w\cortex.cpp\cortex.cpp\engine\utils/command_executor.h(48): error C3861: 'pclose': identifier not found
```

qnixsynapse replied:
I think it is better to use `std::unique_ptr<FILE, int (*)(FILE*)> m_pipe{nullptr, PCLOSE};` which I have used in this PR #2075 and ultimately found out that it is duplicate.


Also, a bit proper formatting is expected.

github-roushan committed:
Update command_executor.h

use macro PCLOSE to correctly map to _pclose on windows

github-roushan committed:
fix typo: revert PCLOSE to _pclose for windows

The PCLOSE macro was mistakenly defined as pclose for Windows. This commit fixes the typo, reverting it back to _pclose

github-roushan commented:
![Screenshot (22)](https://github.com/user-attachments/assets/a9eb5c30-fa62-44ea-b78b-5bd2f1200501)

fixed it @qnixsynapse

github-roushan committed:
Merge branch 'dev' into remove_warn

qnixsynapse commented on engine/utils/command_executor.h:
```diff
@@ -45,5 +45,5 @@ class CommandExecutor {
 
  private:
-  std::unique_ptr<FILE, decltype(&PCLOSE)> m_pipe{nullptr, PCLOSE};
+ std::unique_ptr<FILE, void (*)(FILE*)> m_pipe{nullptr, [](FILE* file) { if (file) { PCLOSE(file); } }};
```
Formatting please

github-roushan committed:
chore: fix formatting

github-roushan commented:
can we push this as well @sangjanai

vansangpfiev committed:
Merge branch 'dev' into remove_warn
